tests/test_job_spec.py

```diff
@@ -12,16 +12,14 @@ def _test_spec(spec: JobSpec) -> None:
 
 
 def test_environment_types() -> None:
-    with pytest.raises(TypeError):
-        _test_spec(JobSpec(environment={'foo': 1}))  # type: ignore
 
     with pytest.raises(TypeError):
-        _test_spec(JobSpec(environment={1: 'foo'}))  # type: ignore
+        _test_spec(JobSpec(executable='true', environment={1: 'foo'}))  # type: ignore
```

Comment:
I would caution against assuming that `/bin` is in `PATH` or that `PATH` is used when launching executables.